src/Utils/Rotation.cpp

```diff
@@ -312,15 +312,335 @@ void utils::Rotation::checkUnitary()
+        {
+            utils::Matrix3d M_adj_Tk;
+
+            /*
+            // row 2 x row 3
+            cross(&(Mk[3]), &(Mk[6]), &M_adj_Tk(0, 0)));
+            // row 3 x row 1
+            cross_product(&(Mk[6]), &(Mk[0]), &(M_adj_Tk[3]));
+            // row 1 x row 2
+            cross_product(&(Mk[0]), &(Mk[3]), &(M_adj_Tk[6]));
+            */
+
+            det = Mk(0, 0) * M_adj_Tk(0, 0) + Mk(0, 1) * M_adj_Tk(0, 1) + Mk(0, 2) * M_adj_Tk(0, 2);
+
+            if( det <= 1e-6 )
+            {
+                use_svd = true;
+                std::cout << "Oups, SVD serait utile!\n";
+                break;
+            }
+
+            if( det == 0.0 )
+            {
+                printf("Warning (polarDecomposition) : zero determinant encountered.\n");
+                break;
+            }
+
+            //mat_adj_T one-norm
+            float MadjT_one_norm = 0.0f;
+            for (int i = 0; i < 3; i++)
+            {
+                float col_abs_sum = fabs(M_adj_Tk(i, 0)) + fabs(M_adj_Tk(i, 1)) + fabs(M_adj_Tk(i, 2));
+                if (col_abs_sum > MadjT_one_norm)
+                    MadjT_one_norm = col_abs_sum;
+            }
+
+            //mat infinity-norm
+            float MadjT_inf_norm = 0.0;
+            for (int i = 0; i < 3; i++)
+            {
+                float row_sum = fabs(M_adj_Tk(i, 0)) + fabs(M_adj_Tk(i, 1)) + fabs(M_adj_Tk(i, 2));
+                if (row_sum > M_inf_norm)
+                    M_inf_norm = row_sum;
+            }
+
+            
+            float gamma = sqrtf(sqrtf((MadjT_one_norm * MadjT_inf_norm) / (M_one_norm * M_inf_norm * det * det)));
+            float g1 = gamma * 0.5f;
+            float g2 = 0.5f / (gamma * det);
+
+            for(int i = 0; i < 3; i++)
+            {
+                for(int j = 0; i < 3; j++)
+                    {
+                        Ek(i, j) = Mk(i, j);
+                        Mk(i, j) = g1 * Mk(i, j) + g2 * M_adj_Tk(i, j);
+                        Ek(i, j) -= Mk(i, j);
+                    }
+            }
+
+            //Ek one-norm
+            float E_one_norm = 0.0f;
+            for (int i = 0; i < 3; i++)
+            {
+                float col_abs_sum = fabs(Ek(i, 0)) + fabs(Ek(i, 1)) + fabs(Ek(i, 2));
+                if (col_abs_sum > E_one_norm)
+                    E_one_norm = col_abs_sum;
+            }
+
+            //M one-norm
+            float M_one_norm = 0.0f;
+            for (int i = 0; i < 3; i++)
+            {
+                float col_abs_sum = fabs(M(i, 0)) + fabs(M(i, 1)) + fabs(M(i, 2));
+                if (col_abs_sum > M_one_norm)
+                    M_one_norm = col_abs_sum;
+            }
+
+            //M infinity-norm
+            float M_inf_norm = 0.0;
+            for (int i = 0; i < 3; i++)
+            {
+                float row_sum = fabs(M(i, 0)) + fabs(M(i, 1)) + fabs(M(i, 2));
+                if (row_sum > M_inf_norm)
+                    M_inf_norm = row_sum;
+            }
```

Comment:
Same comments, but here the definitely seem to be merge. Again, row_sum and col_sum don't seem to differ, which seems odd